addons/hr_calendar/tests/test_event_interval.py

```diff
@@ -0,0 +1,72 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+from datetime import datetime
+from pytz import timezone
+from odoo.addons.hr_calendar.tests.common import TestHrCalendarCommon
+
+
+@tagged('event_interval')
+class TestEventInterval(TestHrCalendarCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.company_A.resource_calendar_id = cls.calendar_35h
+
+    def test_allday_event_during_working_day(self):
+        event = self.env['calendar.event'].with_context(company_id=self.company_A.id).create([
+            {
+                'start': datetime(2024, 7, 12),
+                'stop': datetime(2024, 7, 12, 23, 59, 59),
+                'allday': True,
+                'name': "Event 1"
```

Comment:
' ? R----------------------------------------------

especially if you want to do 

```py
                'tz': "Europe/Brussels",
                'tz': 'Europe/Brussels',
```